extensions/wikia/ArticleVideo/scripts/ooyala-player.js

```diff
@@ -0,0 +1,120 @@
+define('ooyala-player', function () {
+
+	var baseJSONSkinUrl = '/extensions/wikia/ArticleVideo/scripts/ooyala/skin.json?v=2';
+	// TODO ooyala only supports font icons so we probably need to extract our DS icons to font
+	var playIcon = '<svg width="22" height="30" viewBox="0 0 22 30" xmlns="http://www.w3.org/2000/svg"><path d="M21.573 15.818l-20 14c-.17.12-.372.18-.573.18-.158 0-.316-.037-.462-.112C.208 29.714 0 29.372 0 29V1C0 .625.207.283.538.11c.33-.17.73-.146 1.035.068l20 14c.268.187.427.493.427.82 0 .325-.16.63-.427.818z"/></svg>'
+
+	function OoyalaHTML5Player(container, params, onPlayerCreate) {
+		var playerWidth = container.scrollWidth;
+
+		this.params = params;
+		this.params.width = playerWidth;
+		this.params.height = Math.floor((playerWidth * 9) / 16);
+		this.params.onCreate = this.onCreate.bind(this);
+
+		this.onPlayerCreated = onPlayerCreate;
+
+		this.params.skin = {
+			config: baseJSONSkinUrl
+		};
+
+		this.containerId = container.id;
+		this.player = null;
+	}
+
+	OoyalaHTML5Player.prototype.setUpPlayer = function () {
+		this.createPlayer();
+	};
+
+	/**
+	 * @returns {void}
+	 */
+	OoyalaHTML5Player.prototype.createPlayer = function () {
+		var self = this;
+		window.OO.ready(function () {
+			self.player = OO.Player.create(self.containerId, self.params.videoId, self.params);
+		});
+	};
+
+	/**
+	 * @param {*} player
+	 * @returns {void}
+	 */
+	OoyalaHTML5Player.prototype.onCreate = function (player) {
+		var messageBus = player.mb,
+			self = this;
+
+		this.onPlayerCreated(player);
+
+		messageBus.subscribe(window.OO.EVENTS.PLAYBACK_READY, 'ui-update', function () {
```

Comment:
Why not use the `featured-video` [subscriber](http://apidocs.ooyala.com/player_v4_js/OO.MessageBus.html#subscribe)?